workflow/common/serviceaccount.go

```diff
@@ -0,0 +1,72 @@
+package common
+
+import (
+	"github.com/argoproj/argo/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/serviceaccount"
+)
+
+// GetServiceAccountTokenByAccountName returns the name of the first referenced secret which is a ServiceAccountToken for the service account
+func GetServiceAccountTokenByAccountName(clientset kubernetes.Interface, namespace, name string) (*corev1.Secret, error) {
```

Comment:
Just FYI I referenced the Kubernetes code to write the functions in this file.
https://github.com/kubernetes/kubernetes/blob/b8eecd671d7d895ecd4f0fc6930b82e55298e257/plugin/pkg/admission/serviceaccount/admission.go#L327-L349